internal/layerfile/layerfile_test.go

```diff
@@ -83,3 +84,55 @@ func TestToLayers(t *testing.T) {
 	assert.Equal(t, mainTfContent, modelLayers[0].Files[0].Content)
 	assert.Equal(t, 0, len(modelLayers[0].Dependencies))
 }
+
+func TestToLayers_ValidateNameOfLayerDefinitions(t *testing.T) {
+	tests := []struct {
+		name string
+		lf   layerfile
+		err  error
+	}{
+		{
+			name: "Name has spaces",
+			lf: layerfile{
+				Layers: []layerfileLayer{
+					{
+						Name: "invalid name for a layer definition",
+					},
+				},
+			},
+			err: errors.New("invalid name: invalid name for a layer definition"),
+		},
+		{
+			name: "Name has special character",
+			lf: layerfile{
+				Layers: []layerfileLayer{
+					{
+						Name: "invalid!",
+					},
+				},
+			},
+			err: errors.New("invalid name: invalid!"),
+		},
+		{
+			name: "Valid name",
+			lf: layerfile{
+				Layers: []layerfileLayer{
+					{
+						Name: "validname",
+					},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := tt.lf.ToLayers()
+			if err != nil {
+				assert.EqualError(t, tt.err, err.Error())
+			} else {
+				assert.NoError(t, tt.err)
+			}
```

Comment:
The reason why im doing if-else in this test: 

I Kept getting this error when using `assert.Equal(t, tt.err, err)`

```suggestion
			_, err := tt.lf.ToLayers()
			assert.Equal(t, tt.err, err)
```

```
go test -v -run=TestToLayers_ValidateNameOfLayerDefinitions github.com/ergomake/layerform/internal/layerfile
=== RUN   TestToLayers_ValidateNameOfLayerDefinitions
=== RUN   TestToLayers_ValidateNameOfLayerDefinitions/Name_has_spaces
    layerfile_test.go:136: 
                Error Trace:    /home/rafi/Documents/oss/github/layerform/internal/layerfile/layerfile_test.go:136
                Error:          Not equal: 
                                expected: invalid name: invalid name for a layer definition
                                actual  : invalid name: invalid name for a layer definition
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -2,4 +2,5 @@
                                  msg: (string) (len=49) "invalid name: invalid name for a layer definition",
                                - stack: (*errors.stack)((len=3) {
                                -  (uintptr) 0x5b7024,
                                + stack: (*errors.stack)((len=4) {
                                +  (uintptr) 0x5b639d,
                                +  (uintptr) 0x5b724e,
                                   (uintptr) 0x4e89ab,
                Test:           TestToLayers_ValidateNameOfLayerDefinitions/Name_has_spaces
=== RUN   TestToLayers_ValidateNameOfLayerDefinitions/Name_has_special_character
    layerfile_test.go:136: 
                Error Trace:    /home/rafi/Documents/oss/github/layerform/internal/layerfile/layerfile_test.go:136
                Error:          Not equal: 
                                expected: invalid name: invalid!
                                actual  : invalid name: invalid!
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -2,4 +2,5 @@
                                  msg: (string) (len=22) "invalid name: invalid!",
                                - stack: (*errors.stack)((len=3) {
                                -  (uintptr) 0x5b7099,
                                + stack: (*errors.stack)((len=4) {
                                +  (uintptr) 0x5b639d,
                                +  (uintptr) 0x5b724e,
                                   (uintptr) 0x4e89ab,
                Test:           TestToLayers_ValidateNameOfLayerDefinitions/Name_has_special_character
=== RUN   TestToLayers_ValidateNameOfLayerDefinitions/Valid_name
--- FAIL: TestToLayers_ValidateNameOfLayerDefinitions (0.00s)
    --- FAIL: TestToLayers_ValidateNameOfLayerDefinitions/Name_has_spaces (0.00s)
    --- FAIL: TestToLayers_ValidateNameOfLayerDefinitions/Name_has_special_character (0.00s)
    --- PASS: TestToLayers_ValidateNameOfLayerDefinitions/Valid_name (0.00s)
FAIL
FAIL    github.com/ergomake/layerform/internal/layerfile        0.005s
FAIL
```